vscode-dotnet-runtime-library/src/Utils/ExecutableArchitectureDetector.ts

```diff
@@ -0,0 +1,202 @@
+/*---------------------------------------------------------------------------------------------
+*  Licensed to the .NET Foundation under one or more agreements.
+*  The .NET Foundation licenses this file to you under the MIT license.
+*--------------------------------------------------------------------------------------------*/
+
+import { readFileSync } from 'fs';
+
+export type Architecture = 'x86' | 'x64' | 'arm64' | 'other' | null;

```

Comment:
What's `'other'` vs `null` in this context?

Reply:
Checking the wiki page, there are dozens of architecture values possible. There is a test that shows this, with arm32, which we produce an arm32 dotnet. We dont need to worry about those in the context of dotnet/code because it is the intersection of what platforms both support, which is very limited to these values above.